src/test/kotlin/io/imotions/bson4k/BsonTest.kt

```diff
@@ -108,6 +109,17 @@ class BsonTest : StringSpec({
         deserialized shouldBe clazz
     }
 
+    "Encode and decode without serialized default values" {
+        val clazz = Wrapper2Null<Boolean, Boolean>(y = false)
```

Comment:
I can't find the sources for this easily. I recommend making this test more self-contained, i.e., place the sources of the class being serialized here nearby. Or, use a better-named class tailored to this purpose, e.g. `TypeWithNoDefaults`.

Same comment for the other changes. I don't understand what `Wrapper2Null`  is or why this type is relevant in these tests.